### PR TITLE
Update mason documentation

### PR DESCRIPTION
**Pull request by daviditen**
Update mason documentation to include setting MASON_REGISTRY to include
multiple registries including local registries.

Added a section with an example of using a local registry with local library
packages.

Added Bricks/ to the example showing what a registry hierarchy looks like.

Removed redundant 'make clobber' section that is identical to the 'make clean'
section and does exactly the same thing.

Fixed a few typos and made spacing between sections consistent.

### CODE REVIEW

File: doc/rst/tools/mason/mason.rst [hunk 1]
```diff
@@ -287,6 +272,68 @@ Once your package is uploaded, maintain the integrity of your package, and pleas
...
+   mkdir /path/to/local/registry
+   cd /path/to/local/registry
+   mkdir -p Bricks/ProjectA Bricks/ProjectB
+   cp /path/to/my/projects/ProjectA/Mason.toml Bricks/ProjectA/0.1.0.toml
+   cp /path/to/my/projects/ProjectB/Mason.toml Bricks/ProjectB/0.1.0.toml
+
+   <edit Bricks/ProjectA/0.1.0.toml to add>
+   source = "/path/to/my/projects/ProjectA"
+
+   <edit Bricks/ProjectB/0.1.0.toml to add>
```
Comment (ben-albrecht) [inline doc/rst/tools/mason/mason.rst]: These may render better as a `# bash comment`

File: doc/rst/tools/mason/mason.rst [hunk 2]
```diff
@@ -8,8 +8,6 @@ mason
 Mason is Chapel's package manager
```
Comment (ben-albrecht) [inline doc/rst/tools/mason/mason.rst]: A TOC might be nice, given how long this page is getting now, e.g.

```
.. contents:: Table of Contents
```

File: doc/rst/tools/mason/mason.rst [hunk 3]
```diff
@@ -287,6 +272,68 @@ Once your package is uploaded, maintain the integrity of your package, and pleas
...
+starting with Bricks for ``ProjectA`` and ``ProjectB`` which were created with
+``mason new ProjectA`` and ``mason new ProjectB``, and are located at
+``/path/to/my/projects/Project[AB]``. It is expected that mason will be
+extended to simplify and handle more of this process.
+
+First create commit, and tag the projects that will be in the registry:
+
+.. code-block:: sh
+
+   cd /path/to/my/projects
```
Comment (ben-albrecht) [inline doc/rst/tools/mason/mason.rst]: Adding some concise comments on their own line like `# Create project` might help break this up a little and improve readability.

File: doc/rst/tools/mason/mason.rst [hunk 4]
```diff
@@ -287,6 +272,68 @@ Once your package is uploaded, maintain the integrity of your package, and pleas
...
+================
+
+It is sometimes desirable to use a local registry, for example with libraries
+you don't intend to distribute. The following steps create a local registry
+starting with Bricks for ``ProjectA`` and ``ProjectB`` which were created with
+``mason new ProjectA`` and ``mason new ProjectB``, and are located at
+``/path/to/my/projects/Project[AB]``. It is expected that mason will be
+extended to simplify and handle more of this process.
+
+First create commit, and tag the projects that will be in the registry:
```
Comment (ben-albrecht) [inline doc/rst/tools/mason/mason.rst]: Missing comma, I think:

"First create, commit, and tag the projects ..."

File: doc/rst/tools/mason/mason.rst [hunk 5]
```diff
@@ -19,35 +17,29 @@ In ``$CHPL_HOME`` run the following:
...
-  Defaults to ``$HOME/.mason``.
-- ``MASON_REGISTRY`` : A valid git URL pointing to a repository of package information.
-  Defaults to ``https://github.com/chapel-lang/mason-registry``
+- ``MASON_HOME`` : Path to a directory where mason will store cached registry
+  and package data. Defaults to ``$HOME/.mason``.
+- ``MASON_REGISTRY`` : A comma separated list of ``name|location`` pairs, where
+  ``name`` is a local name for the registry at ``location``. Defaults to
+  ``mason-registry|https://github.com/chapel-lang/mason-registry``. If the
+  ``name|`` part of a pair is omitted it is inferred to be the word following
+  the final slash in ``location``.
```
[the remark this answers is not in the document]
Comment (daviditen) [reply doc/rst/tools/mason/mason.rst]: I forgot about that.  I'll add it.

File: doc/rst/tools/mason/mason.rst [hunk 6]
```diff
@@ -19,35 +17,29 @@ In ``$CHPL_HOME`` run the following:
...
 It builds the mason binary so that the command line interface can be used.
 This installs mason in the same place as the chapel compiler (``chpl``) so that
 mason can be used anywhere in the user's file system.
 
-
 To remove mason, change directory to ``$CHPL_HOME/tools/mason`` and run:
 
 .. code-block:: sh
 
    make clean
```
[the remark this answers is not in the document]
Comment (daviditen) [reply doc/rst/tools/mason/mason.rst]: There is no difference.  The clobber rule is simply:

```
clobber: clean
```

**Comment by daviditen** [comment]
Thanks for reviewing @ben-albrecht!  I've addressed everything except the TOC in the new commit.  I'll work on the TOC next.

File: doc/rst/tools/mason/mason.rst (hunk 2 above)
Comment (daviditen) [reply doc/rst/tools/mason/mason.rst]: Oh.  I didn't realize that the code you gave was all there was to it.  I was imagining having to create all sorts of RST links between the TOC and the sections.

**Comment by daviditen** [comment]
I added a TOC, so I think that I've covered all of your comments.  Thanks!

**Pull request merged**